src/main/java/de/tum/in/www1/artemis/service/QuizSubmissionService.java

```diff
@@ -215,34 +221,20 @@ public QuizSubmission saveSubmissionForExamMode(QuizExercise quizExercise, QuizS
             throw new EntityNotFoundException("Participation for quiz exercise " + quizExercise.getId() + " and quiz submission " + quizSubmission.getId() + " for user "
                     + user.getLogin() + " was not found!");
         }
-        StudentParticipation studentParticipation = optionalParticipation.get();
-        quizSubmission.setParticipation(studentParticipation);
-        // remove result from submission (in the unlikely case it is passed here), so that students cannot inject a result
-        quizSubmission.setResults(new ArrayList<>());
-        quizSubmissionRepository.save(quizSubmission);
-
-        // versioning of submission
-        try {
-            submissionVersionService.saveVersionForIndividual(quizSubmission, user);
-        }
-        catch (Exception ex) {
-            log.error("Quiz submission version could not be saved", ex);
-        }
-
-        log.debug("submit exam quiz finished: {}", quizSubmission);
-        return quizSubmission;
+        return optionalParticipation.get();
     }
 
     /**
-     * Returns true if student has submitted at least once for the given quiz batch
+     * Save the quiz submission to the database and also save the submission version
```

Comment:
Maybe add a comment why the participation is set here? Calling this method it is not clear why you would have to provide the quizExercise and the user to save the quizSubmission.

Reply:
Yes, it happens in AbstractQuizSubmissionService#saveSubmissionForExamMode(). Updated the JavaDoc